File: 27-remove-element/remove-element.go
// Problem: https://leetcode.cn/problems/remove-element/
// Accepted at: 2025年3月4日 09:04

func removeElement(nums []int, val int) int {
    p1:=0
    p2:=len(nums)-1
    k:=len(nums)
    for p1<=p2{
        if nums[p2]==val{
            p2--
            k--
            continue
        }
        if nums[p1]==val{
            nums[p1]=nums[p2]
            p2--
            k--
            continue
        }else{
            p1++
        }
    }
    return k
}

File: 27-remove-element/remove-element.py
# Problem: https://leetcode.cn/problems/remove-element/
# Accepted at: 2026年8月29日 16:55

class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        k = 0
        j = 0
        for i in range(len(nums)):
            if nums[i] == val:
                i = i + 1
            else:
                nums[j] = nums[i]
                i = i + 1
                j = j + 1
                k = k + 1
        return k
